docs/2.build/4.web3-apps/chain-ui.md

```diff
@@ -0,0 +1,77 @@
+---
+id: chain-deployed-ui
+title: Chain Deployed UI
+sidebar_label: Chain Deployed UI
+---
+
+[Chain Deployed UI](https://github.com/near/chain-deployed-ui) is a [decentralized frontend solution](frontend.md#decentralized-frontend-solutions) that lets you to deploy dApp frontends directly on the NEAR blockchain. A frontend can be built with typical Web2 tooling (e.g. React + Vite + node.js dependencies), and the resulting compressed bundle can be stored in the state of a smart contract.
+
+## Overview
+
+Thanks to the fact that storage on NEAR is cheap, it's feasible to host small applications on chain, especially if some optimization is added to reduce redundant code between deployed dApps. This offers a straightforward path to decentralized hosting as an alternative to static deployment platforms such as Vercel, GitHub Pages, Render, and others.
+
+The Chain Deployed UI tooling provides the boilerplate app code (React, Vue, TypeScript), along with the `FileStore` smart contract and basic CDN tools so you can build a dApp, pack it, and store it on chain.
+
+This `FileStore` smart contract deployed by NEAR is designed to store and serve dApp bundles for any user, so you don't need to have an understanding of smart contract development. You only need to understand how to sign a transaction (with help from our tooling) and to have enough NEAR tokens to pay for storage.
+
+:::tip
+
+You can still deploy your own file storage smart contract if you want to have additional customization or if you want total control over your data.
+
+:::
+
+### Pros
+
+- Decentralized front-ends are less susceptible to single points of failure, making them more resistant to attacks and server downtimes
```

Comment:
I think this glosses over the fact that our solution is entirely contingent on hosting a web service that is vulnerable to these exact things 😅

Reply:
Right, unless they have a setup with multiple gateways and RPC servers, they could still have a single point of failure.

I'm removing this point for now.